promote-to-stable/action.yaml

```diff
@@ -34,12 +34,39 @@ runs:
       shell: bash
       run: |
         sudo snap install --classic snapcraft
+    
+    - name: Find the snapcraft.yaml path
+      id: yaml-path
+      shell: bash
+      run: |
+        if [[ -n "${{ inputs.snapcraft-yaml-path }}" ]]; then
+          yaml_path="${{ inputs.snapcraft-yaml-path }}"
+        else
+          snapcraft_yaml_paths=(
+            "snap/snapcraft.yaml"
+            "snapcraft.yaml"
+            "build-aux/snap/snapcraft.yaml"
+            ".snapcraft.yaml"
+          )
+          
+          for file in "${snapcraft_yaml_paths[@]}"; do
+              if [[ -f "$file" ]]; then
+                  yaml_path="$file"
+              fi
+          done
+        fi
+        if [[ ! -n "${yaml-path}" ]]; then
+           echo "No snapcraft.yaml found" >2
+           exit 1
+        fi
+        echo "yaml-path=${yaml_path}" >> "$GITHUB_OUTPUT"
```

Comment:
Fix

Reply:
This one is actually correct - the output of the step can in fact be `yaml-path`. I'm about to add some commits with linting and a styleguide